### PR TITLE
Updates to minimized repo and style and consistency changes

### PR DESCRIPTION
**Pull request by ggantos**
Made large deletions and mostly worked in evml/keras directory. Edited all files for docstrings, imports, unused variables, and PEP8 consistency.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -9,12 +9,11 @@
 url = "https://github.com/ai2es/" + project_name
 # Classifiers listed at https://pypi.org/classifiers/
 classifiers = ["Programming Language :: Python :: 3"]
-setup(name="evidential", # Change
+setup(name="evidential",  # Change
       version=version,
       description=package_description,
       url=url,
       author="John Schreck",
```
Comment (jsschreck) [inline setup.py]: Can you update the author list (probably the four of us for now, maybe other people will contribute later)

File: evml/torch/regression_losses.py [hunk 2]
```diff
@@ -1,20 +1,12 @@
+""" Torch losses for regression models """
+
 import numpy as np
 import torch
-from torch import nn
 import torch.nn.functional as F
```
Comment (jsschreck) [inline evml/torch/regression_losses.py:7]: Can you add some references here, similar to how you did it for the keras code pointing to Amini?

https://github.com/hxu296/torch-evidental-deep-learning/blob/main/evidential_deep_learning/losses/continuous.py

https://github.com/dougbrion/pytorch-classification-uncertainty/blob/master/losses.py

And the updated version to Amini's original -- https://github.com/deargen/MT-ENet/blob/468822188f52e517b1ee8e386eea607b2b7d8829/mtevi/mtevi.py

File: evml/torch/mc_dropout.py [hunk 3]
```diff
@@ -1,9 +1,8 @@
 import sys
 import numpy as np
 import torch
-import torch.nn as nn
 import torch.nn.functional as F
-from evml.class_losses import relu_evidence
+from evml.torch.class_losses import relu_evidence
```
Comment (jsschreck) [inline evml/torch/mc_dropout.py:6]: I don't know that we necessarily need to point to stack, but most of this code came from the given answer here: https://stackoverflow.com/questions/63285197/measuring-uncertainty-using-mc-dropout-on-pytorch

**Review by jsschreck — CHANGES_REQUESTED** [review]
Overall the simplification of the repo looks very nice. Most of my comments were directed at making sure we reference where some of the torch code snippets came from (I left the links in my comments). 

The "main" comment I have about the structure is whether we should keep the torch scripts "metrics.py" and "trainer.py". It might make more sense if we use a notebook to show a torch example, similar to whats in the example for keras. The code in metrics.py seems to be specific to p-type data as well (and not the regression example thats currently in config).

File: environment.yml [hunk 4]
```diff
@@ -2,29 +2,29 @@ name: evidential
 channels:
   - conda-forge
 dependencies:
-  - python=3.8
-  - numpy
+  - python =3.8
+  - numpy ~=1.23.5
```
Comment (djgagne) [inline environment.yml]: What does the `~=` symbol mean in the environment file? I would prefer not to specify library versions unless absolutely necessary (e.g., tensorflow probability only works with specific versions of tensorflow and each version of tensorflow has a limited supported range of versions with numpy) because it can cause conda dependency conflicts down the line.

File: evml/keras/losses.py [hunk 5]
```diff
@@ -1,16 +1,13 @@
 import tensorflow as tf
-from tensorflow.keras import backend as K
 import numpy as np
-from tensorflow.keras.utils import get_registered_name
 
 
 class DirichletEvidentialLoss:
```
Comment (djgagne) [inline evml/keras/losses.py]: Should DirichletEvidentialLoss inherit from tensorflow.keras.losses.Loss like EvidentialRegressionLoss does?

File: evml/keras/models.py [hunk 6]
```diff
@@ -1,49 +1,132 @@
...
+    def fit(self, x, y):
+        inputs = x.shape[1]
+        if len(y.shape) == 1:
+            outputs = 1
+        else:
+            outputs = y.shape[1]
+        self.build_neural_network(inputs, outputs)
+        self.model.fit(x, y, batch_size=self.batch_size, epochs=self.epochs, verbose=self.verbose, shuffle=True)
+        self.training_var = np.var(y)
+        tf.keras.models.save_model(self.model, self.save_path+'/test_model/model.h5', save_format='h5')
```
Comment (djgagne) [inline evml/keras/models.py]: I suggest moving the save_model command out of the fit function and into a separate save_model method. This could cause hard-to-trace bugs for multiple reasons.
1. If `self.save_path` does not have a `/` at the end, you will get a path error.
2. If the test_model directory does not exist, you will get an OS Error. 
3. model.h5 is a very generic name. We should have some ability to customize model names or use something that produces a unique model id based on when the model was trained.

When constructing a path for saving anything, use os.path.join to link multiple sub-paths together into a larger path. It automatically handles the presence or absence of slashes.

File: evml/keras/models.py [hunk 7]
```diff
@@ -1,49 +1,132 @@
...
-    aleatoric = prob - prob**2 - epistemic
-    return prob, u, aleatoric, epistemic
\ No newline at end of file
+    def predict(self, x, scaler=None):
+        
+        y_out = self.model.predict(x, batch_size=self.batch_size)
+        if self.uncertainties:
+            return self.calc_uncertainties(y_out, scaler)
+        else:
+            return y_out
```
Comment (djgagne) [inline evml/keras/models.py]: For code reading purposes, it is usually preferable to have one return statement per function or method. In this case, you would assign self.calc_uncertainties to a variable called y_out_final if self.uncertainties is True and otherwise assign y_out to y_out_final. Then you would return y_out_final instead of y_out.

**Review by djgagne — CHANGES_REQUESTED** [review]
I've requested a couple of changes to the EvidentialClassifierDNN model.

File: evml/keras/models.py (hunk 7 above)
Comment (ggantos) [reply evml/keras/models.py]: edited.

File: evml/keras/models.py (hunk 6 above)
Comment (ggantos) [reply evml/keras/models.py]: edited.

File: evml/keras/losses.py (hunk 5 above)
Comment (ggantos) [reply evml/keras/losses.py]: edited.

File: environment.yml (hunk 4 above)
Comment (ggantos) [reply environment.yml]: It specifies a range of version types. I removed them for now and will test the new environment.

File: setup.py (hunk 1 above)
Comment (ggantos) [reply setup.py]: Edited.

File: evml/torch/regression_losses.py (hunk 2 above)
Comment (ggantos) [reply evml/torch/regression_losses.py:7]: Added!

File: evml/torch/mc_dropout.py (hunk 3 above)
Comment (ggantos) [reply evml/torch/mc_dropout.py:6]: Added!

**Review by djgagne — APPROVED** [review]
The most recent changes addressed my comments. Merging.
